code/modules/cm_marines/equipment/sentries.dm

```diff
@@ -931,16 +1063,47 @@
 		var/mob/living/carbon/human/H = M
 		if(istype(H) && H.get_target_lock(iff_signal)) continue
 
+
+
 		var/angle = get_dir(src, M)
-		if(angle & dir)
-			path = getline2(src, M)
+		if(angle & dir || radial_mode)
+			path = getline2(src, M, TRUE)
+			if(alerts_on) //They're within our field of detection and thus can trigger the alarm
+				if(world.time > (last_alert + SENTRY_ALERT_DELAY) || !(M in alert_list)) //if we're not on cooldown or the target isn't in the list, sound the alarm
+					playsound(loc, 'sound/machines/warning-buzzer.ogg', 50, FALSE)
+					sentry_alert(SENTRY_ALERT_HOSTILE, M)
+					alert_list.Add(M)
+					last_alert = world.time
 		else
 			continue
 
 		if(path.len)
+			var/blocked = FALSE //reset blocked
+			var/current_turf = get_turf(src)
 			for(T in path)
-				if(T.opacity) continue
-			targets += M
+				if(T == current_turf) //Don't deal with the current turf
+					continue
+				if(T.opacity || T.density)
+					blocked = TRUE
+					continue //don't pass on targets we can't actually fire at.
+
+				if(!blocked) //allows us to stop checks and thus save resources if we find something that blocks LoS/LoF
```

Comment:
There is a `continue`, after `blocked = TRUE`, so this check will never fail and is thus not necessary.